src/Io/MultipartParser.php

```diff
@@ -134,14 +139,42 @@ private function parseFile($headers, $body)
         $this->request = $this->request->withUploadedFiles($this->extractPost(
             $this->request->getUploadedFiles(),
             $this->getFieldFromHeader($headers['content-disposition'], 'name'),
-            new UploadedFile(
-                Psr7\stream_for($body),
-                strlen($body),
-                UPLOAD_ERR_OK,
+            $this->parseUploadedFile($headers, $body)
+        ));
+    }
+
+    private function parseUploadedFile($headers, $body)
+    {
+        $bodyLength = strlen($body);
+
+        $parsedBody = $this->request->getParsedBody();
+        if ($this->maxFileSize === null && is_array($parsedBody)) {
+            $postKeys = array_keys($parsedBody);
+            foreach ($postKeys as $key) {
+                if (strtoupper($key) === 'MAX_FILE_SIZE') {
+                    $this->maxFileSize = $parsedBody[$key];
+                    break;
+                }
+            }
+        }
```

Comment:
If i remember correctly, it's not that easy. According to http://php.net/manual/en/features.file-upload.post-method.php#example-383, the field must precede the file input field.

The reason is, that you can have multiple `MAX_FILE_SIZE` hidden fields. Each field controls the max size of the following file fields.

```html
<input name="file1" type="file"> 
<input name="file2" type="file">
<!-- "file1" and "file2" will have no MAX_FILE_SIZE (no limit) -->

<input name="MAX_FILE_SIZE" value="1024" type="hidden">
<input name="file3" type="file">
<input name="file4" type="file">
<!-- "file3" and "file4" will have MAX_FILE_SIZE = 1024 bytes -->

<input name="MAX_FILE_SIZE" value="0" type="hidden">
<input name="file5" type="file">
<input name="file6" type="file">
<!-- "file5" and "file6" will have MAX_FILE_SIZE = 0 (no limit) -->
```

Reply:
Updated the middleware to support this behavior :+